ui/src/hosts/components/HostsTable.js

```diff
@@ -136,22 +135,32 @@ const HostRow = React.createClass({
       name: PropTypes.string,
       cpu: PropTypes.number,
       load: PropTypes.number,
+      deltaUptime: PropTypes.number.required,
       apps: PropTypes.arrayOf(PropTypes.string.isRequired),
     }),
   },
 
   shouldComponentUpdate(nextProps) {
-    return shallowCompare(this, nextProps);
+    return this.props.host !== nextProps.host;
```

Comment:
This could be okay if we use it right, but it's a bit dangerous IMO
Say`const host = {name: "Will", load: 99};`
then `HostRow` receives that prop on one render...
then later we do `Object.assign(host, {load: 0, apps: [...]})`
this `shouldComponentUpdate` will prevent the new data from appearing, since this.props.host === nextProps.host

Hence, shallowCompare. Like I said, this may work out. Would be interested to know why this was changed.